unified-runtime/source/adapters/offload/enqueue.cpp

```diff
@@ -28,22 +28,27 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunch(
   //
 
   (void)pGlobalWorkOffset;
-  (void)pLocalWorkSize;
 
-  if (workDim == 1) {
-    std::cerr
-        << "UR Offload adapter only supports 1d kernel launches at the moment";
-    return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+  if (!pLocalWorkSize) {
+    // TODO: This is not optimal, but it is legal
+    static size_t DefaultWorkSize[3] = {1, 1, 1};
+    pLocalWorkSize = DefaultWorkSize;
+  }
+
+  if (pLocalWorkSize[0] > pGlobalWorkSize[0] ||
+      pLocalWorkSize[1] > pGlobalWorkSize[1] ||
+      pLocalWorkSize[2] > pGlobalWorkSize[2]) {
```

Comment:
should do this in a loop based on `workDim` otherwise it's going to segfault if you pass a 2 element array

Reply:
... But I need to fix it for pGlobalWorkSize as well. One sec.